### ISSUE TITLE
Add .clear() method to Selections

### ISSUE THREAD

**Issue by RyanMullins** [PAIR-code/megaplot#11]
tl;dr -- There should be an API that guarantees immediate removal of items in a Selection, and therefore the rendered Scene, in the next animation frame. 

## Background 

In LIT, I'm using multiple [`Selection`s](https://github.com/PAIR-code/megaplot/blob/main/src/lib/selection-types.d.ts#L33) to act as proxies for layers in the [`Scene`](https://github.com/PAIR-code/megaplot/blob/main/src/lib/scene.ts#L29). One of these `Selection`s is always bound to the entire dataset, and the other `Selection`s are bound to some subset of the dataset (not that it matters for this issue, but one Selection can contain [0, N] datums and the other 3 can contain [0, 1] datums). For the `Selection`s containing subsets, I assumed that calling [`.bind()`](https://github.com/PAIR-code/megaplot/blob/main/src/lib/selection-types.d.ts#L73) with an empty array would mark all existing datums (and their associated Sprites) for removal from the `Scene`, effectively clearing the `Selection` and allowing me to make a successive call to `.bind()` with the new contents that would render the data I now cared about in the `Scene`. Some example code below.

```javascript
const data = [/* let's assume this is populated */];

const container = document.querySelector('some_query');
const scene = new Scene({container});
const allData = scene.createSelection();
const someData = scene.createSelection();

allData.bind(data);

function update(someSentinel) {
  someData.bind([]);
  const interestingData = data.filter(d => /* some predicate */);
  someData.bind(interestingData);
}
```

This assumption was incorrect because of how the [`Selection.bind()`](https://github.com/PAIR-code/megaplot/blob/main/src/lib/selection-impl.ts#L103) uniquely identifies binding tasks. Calls to `.bind()` are [transformed into `bindingTask`s](https://github.com/PAIR-code/megaplot/blob/main/src/lib/selection-impl.ts#L270) that are uniquely identified by the `Selection` on which `.bind()` is being called. In the example above, the successive calls to `.bind()` [effectively overwrite each other in the scheduler](https://github.com/PAIR-code/megaplot/blob/main/src/lib/selection-impl.ts#L293) because they process before the next animation frame happens.

## Desired Solution

It would be useful if `Selection` has a `.clear()` API that guarantees the immediate execution of an [`exitTask`](https://github.com/PAIR-code/megaplot/blob/main/src/lib/selection-impl.ts#L211)-like action that removes all content from the `Selection`. This would allow effective chaining of effects for the `Selection`s-as-layers use case. See revised update function below.

```javascript
function update(someSentinel) {
  someData.clear();
  const interestingData = data.filter(d => /* some predicate */);
  someData.bind(interestingData);
}
```

**Comment by jimbojw** [comment]
Implemented in #14
